k2/csrc/fsa.h

```diff
@@ -159,18 +166,50 @@ struct Cfsa {
   Arc *arcs;   // Note: arcs[BeginArcIndex()] through arcs[EndArcIndex() - 1]
                // are valid.
 
-  // Constructor from Fsa
-  Cfsa(const Fsa &fsa);
+  Cfsa();
+  // Constructor from Fsa. The passed `fsa` should be kept alive
+  // as long as this cfsa is alive.
+  explicit Cfsa(const Fsa &fsa);
 
   Cfsa &operator = (const Cfsa &cfsa) = default;
   Cfsa(const Cfsa &cfsa) = default;
 
   int32_t NumStates() const { return num_states; }
-  int32_t FinalState() const { return num_states - 1; }
+  int32_t NumArcs()  const { return end_arc - begin_arc; }
+  int32_t FinalState() const {
+    DCHECK_GE(num_states, 2) << "It's an error to invoke this method for "
```

Comment:
I guess using `CHECK_GE` instead of `DCHECK` is better here, as users usually would not call `FinalState` too many times on one cfsa, there is no efficiency issue here (but release check will make sure the program can stop early when there's error)

Reply:
fixed.